docs/dev/Pagination-v2.md

```diff
@@ -285,3 +285,87 @@ OpenSearchExecutionEngine->>+ProjectOperator: getTotalHits
     ResourceMonitorPlan-->>-ProjectOperator: value
   ProjectOperator-->>-OpenSearchExecutionEngine: value
 ```
+
+### Plan Tree changes
+
+#### Abstract Plan tree
```

Comment:
consider adding a state diagram to say both pagination and non-pagination both follow the following execution: 
```
<Abstract Plan> -> <Logical Plan> -> <Optimized Logical Plan> -> <Physical Plan> -> <Request>
```

We should seriously consider following the language used in https://user-images.githubusercontent.com/46505291/203645359-3f2fff73-a210-4bc0-a582-951a27de684d.jpg
* under docs/dev/query-optimizer-improvement.md Examples section.